dicesGame/src/test/java/com/example/dices/services/DiceServiceTest.java

```diff
@@ -0,0 +1,240 @@
+        // verify that the service returns the expected list
+        assertEquals(diceModel.size(), result.size());
+        assertEquals(diceModel.get(0).getDiceId(), result.get(0).getDiceId());
+        assertEquals(diceModel.get(0).getDiceSize(), result.get(0).getDiceSize());
+        assertEquals(diceModel.get(1).getDiceId(), result.get(1).getDiceId());
+        assertEquals(diceModel.get(1).getDiceSize(), result.get(1).getDiceSize());
+        assertEquals(diceModel.get(2).getDiceId(), result.get(2).getDiceId());
+        assertEquals(diceModel.get(2).getDiceSize(), result.get(2).getDiceSize());
+        verify(diceRepository).findAll();
+    }
+
+
+    @Test
+    void testCreateDiceWithValidSize() {
+        // ARRANGE
+        DiceModel validDice = new DiceModel();
+        validDice.setDiceSize(TestServiceConstants.DEFAULT_DICE_SIZE1);
+
+        // ACT
+        when(diceRepository.save(validDice)).thenReturn(validDice);
+        DiceModel createdDice = diceService.createDice(validDice);
+
+        // ASSERT
+        // Verify that the service method returns the expected result
+        assertThat(createdDice).isNotNull();
+
+        // Verify that the diceRepository.save() method was called with the inputDice
+        verify(diceRepository, times(1)).save(validDice);
+
+    }
+
+
+    @Test
+    void testCreateDiceWithInvalidSize() {
+        // ARRANGE
+        DiceModel invalidDice = new DiceModel();
+        invalidDice.setDiceSize(TestServiceConstants.DEFAULT_DICE_INVALID_SIZE);
+
+        // ACT
+        DiceModel createdDice = diceService.createDice(invalidDice);
+
+        // ASSERT
+        // validates that the "save" method of the repository was not called.
+        verify(diceRepository, never()).save(invalidDice);
+
+        // validates that the error message is generated
+        assertNull(createdDice);
+
+    }
+
+
+    @Test
+    void testGetByIdExist() {
+        // ARRANGE
+        int diceId = TestServiceConstants.DEFAULT_DICE_ID1;
+
+        DiceModel retrieveDice = new DiceModel();
+        retrieveDice.setDiceId(diceId);
+        retrieveDice.setDiceSize(TestServiceConstants.DEFAULT_DICE_SIZE1);
+
+        // mock the behavior of the diceRepository.findById() method
+        when(diceRepository.findById(diceId)).thenReturn(Optional.of(retrieveDice));
+
+        // ACT
+        DiceModel obtainedDice = diceService.getById(diceId);
+
+        // ASSERT
+        verify(diceRepository).findById(diceId);
+        assertEquals(diceId, obtainedDice.getDiceId());
+        assertEquals(TestServiceConstants.DEFAULT_DICE_SIZE1, obtainedDice.getDiceSize());
+
+    }
+
+
+    @Test
+    void testGetByIdNotExist() {
+        // ARRANGE
+        int diceId = TestServiceConstants.DEFAULT_DICE_NOT_EXIST;
+
+        // Mock the behavior of the diceRepository.findById() method
+        when(diceRepository.findById(diceId)).thenReturn(Optional.empty());
+
+        // ACT
+        DiceModel obtainedDice = diceService.getById(diceId);
```

Comment:
You are missing making an assertion on this variable.

Reply:
I have included: 
 assertTrue(diceRepository.findById(diceId).isEmpty());